cmd/gateway-main.go

```diff
@@ -94,7 +94,7 @@ func newGatewayLayer(backendType, accessKey, secretKey string) (GatewayLayer, er
 //
 // DO NOT save this config, this is meant to be
 // only used in memory.
-func newGatewayConfig(accessKey, secretKey, region string) error {
```

Comment:
why is this changed? this is needed when we bring the S3 backend it was kept as a placeholder so that region is addressed properly for S3 backends. 

Please keep it the way it was and only move `createConfigDir()` outside.